kolibri/plugins/coach/assets/src/state/getters/reportUtils.js

```diff
@@ -0,0 +1,69 @@
+
+const ReportConstants = require('../../reportConstants');
+const CoreConstants = require('kolibri.coreVue.vuex.constants');
+
+const ContentNodeKinds = CoreConstants.ContentNodeKinds;
+
+
+/* given an array of objects sum the keys on those that pass the filter */
+function _sumOfKeys(array, key, filter = () => true) {
```

Comment:
I notice that this function is only used twice in the foregoing - a slight pre-abstraction, perhaps?